src/js/controller/viewMixin/core.js

```diff
@@ -175,9 +175,14 @@ var Core = {
                     }
 
                     ymd = datetime.format(viewModel.getStarts(), 'YYYYMMDD');
-                    endTime = viewModel.hasMultiDates ?
-                        datetime.end(viewModel.getEnds()) :
-                        datetime.convertStartDayToLastDay(viewModel.getEnds());
+                    isSameDate = datetime.isSameDate(viewModel.getStarts(), viewModel.getEnds());
```

Comment:
`viewModel.getEnds()` is repeated. It can be cached.

Reply:
also `viewModel.getStarts()` can be cached